src/IO/H5/Python/File.cpp

```diff
@@ -1,6 +1,7 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
 
+#include <boost/algorithm/string/join.hpp>
```

Comment:
Since you depend on `boost` you must also add `boost::boost` to src/IO/H5/Python/CMakeLists.txt